src/inmanta/server/agentmanager.py

```diff
@@ -251,6 +252,12 @@ async def start(self) -> None:
             await self._expire_all_sessions_in_db()
 
         self.add_background_task(self._process_session_listener_actions())
+        # Schedule cleanup agentprocess and agentinstance tables
+        agent_process_purge_interval = opt.agent_process_purge_interval.get()
+        if agent_process_purge_interval > 0:
+            self.schedule(
+                self._purge_agent_processes, interval=agent_process_purge_interval, initial_delay=0, cancel_on_stop=False
+            )
```

Comment:
Why was this moved ?

Reply:
Because I think it should have been in the agent manager in the first place: it is related to management of agents, so it should be in the agent manager.